src/ConfigEditor.tsx

```diff
@@ -32,134 +29,96 @@ export function ConfigEditor(props: Props) {
   const fetchDatabases = async () => {
     const loadedDatabases: string[] = await getBackendSrv().post(resourcesURL + '/databases', {
-      region: defaultKey,
-      catalog: catalog || defaultKey,
+      region: props.options.jsonData.defaultRegion,
+      catalog: props.options.jsonData.catalog,
     });
     return loadedDatabases;
   };
   // Workgroups
-  const [workgroup, setWorkgroup] = useState<string | null>(jsonData.workgroup || null);
   const fetchWorkgroups = async () => {
     const loadedWorkgroups: string[] = await getBackendSrv().post(resourcesURL + '/workgroups', {
-      region: defaultKey,
+      region: props.options.jsonData.defaultRegion,
     });
     return loadedWorkgroups;
   };
-  // OuputLocation
-  const [outputLocation, setOutputLocation] = useState<string | null>(jsonData.outputLocation || null);
 
   const onOptionsChange = (options: DataSourceSettings<AthenaDataSourceOptions, AthenaDataSourceSecureJsonData>) => {
-    // clean up related state
     setSaved(false);
-    setCatalog(null);
-    setDatabase(null);
-    setWorkgroup(null);
     props.onOptionsChange(options);
   };
 
-  const onCatalogChange = (catalog: string | null) => {
-    setCatalog(catalog);
+  const onChange = (resource: ResourceType) => (e: SelectableValue<string> | null) => {
+    const value = e?.value ?? '';
     props.onOptionsChange({
       ...props.options,
       jsonData: {
         ...props.options.jsonData,
-        catalog: catalog || '',
+        [resource]: value,
       },
     });
   };
 
-  const onDatabaseChange = (database: string | null) => {
-    setDatabase(database);
+  const onChangeOutputLocation = (e: FormEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
     props.onOptionsChange({
       ...props.options,
       jsonData: {
         ...props.options.jsonData,
-        database: database || '',
+        outputLocation: value,
       },
     });
   };
 
-  const onWorkgroupChange = (workgroup: string | null) => {
-    setWorkgroup(workgroup);
-    props.onOptionsChange({
-      ...props.options,
-      jsonData: {
-        ...props.options.jsonData,
-        workgroup: workgroup || '',
-      },
-    });
-  };
-
-  const onChangeOutputLocation = (outLoc: string | null) => {
-    setOutputLocation(outLoc);
-    props.onOptionsChange({
-      ...props.options,
-      jsonData: {
-        ...props.options.jsonData,
-        outputLocation: outLoc || '',
-      },
-    });
-  };
-
-  const commonProps = {
-    title: jsonData.defaultRegion ? '' : 'select a default region',
-    disabled: !jsonData.defaultRegion,
-    labelWidth: 28,
-    className: 'width-30',
-  };
   return (
     <div className="gf-form-group">
       <ConnectionConfig {...props} onOptionsChange={onOptionsChange} />
-      <h3>Athena Details </h3>
-      <AthenaResourceSelector
-        resource="catalog"
-        onChange={onCatalogChange}
+      <h3>Athena Details</h3>
+      <ConfigSelect
+        {...props}
```

Comment:
I feel like I usually see {...props} go last in the list in case you want to override a default prop, what do you think, would that be a bad idea here for some reason?

Reply:
In a library, it's more convenient to have `{...props}` at the end last, so people can override them. But here, as a user, I want to be sure that my values are set (and not overridden) so it should be set before. For example, if by any reason, the `props` contain a `.value` I want to use the value defined here, not the default one.